src/dmd/json.d

```diff
@@ -793,19 +793,62 @@ public:
         jsonProperties(d);
         objectEnd();
     }
+
+    private void generateBuildInfo()
+    {
+        objectStart();
+        property("kind", "buildInfo");
+        property("binary", global.params.argv0);
+        property("version", global._version);
+        property("config", global.inifilename ? global.inifilename : null);
+        if (global.params.lib) {
+            property("library", global.params.libname);
+        }
+        objectEnd();
+    }
+    private void generateSemantics()
+    {
+        objectStart();
+        property("kind", "semantics");
+        propertyStart("modules");
+        arrayStart();
+        foreach (m; Module.amodules)
+        {
+            objectStart();
+            if(m.md)
+                property("name", m.md.toChars());
```

Comment:
how about:
property("name", m.md ? m.md.toChars() : "");
to make each entry have each field?

Reply:
I was following the convention established here (https://github.com/dlang/dmd/pull/7757/files/a90c983d6e436e46dde3b1ee07fdd280ea1396e7#diff-1d73c979f233c43e47fb88fac32c21b3L475)

I don't really see much difference whether or not there is an empty name or no name at all, but following the existing convention seems the better choice.